src/transform/index.ts

```diff
@@ -1,165 +1,137 @@
-import { OperationObject, PathItemObject, SchemaFormatter } from "../types";
+import { GlobalContext, OperationObject, PathItemObject } from "../types";
 import { comment, tsReadonly } from "../utils";
 import { transformHeaderObjMap } from "./headers";
 import { transformOperationObj } from "./operation";
 import { transformPathsObj } from "./paths";
-import { transformResponsesObj, transformRequestBodies } from "./responses";
+import { transformRequestBodies } from "./request";
+import { transformResponsesObj } from "./responses";
 import { transformSchemaObjMap } from "./schema";
 
-interface TransformOptions {
-  formatter?: SchemaFormatter;
-  immutableTypes: boolean;
-  rawSchema?: boolean;
-  version: number;
-}
-
-export function transformAll(schema: any, { formatter, immutableTypes, rawSchema, version }: TransformOptions): string {
-  const readonly = tsReadonly(immutableTypes);
+export function transformAll(schema: any, ctx: GlobalContext): Record<string, string> {
+  const readonly = tsReadonly(ctx.immutableTypes);
 
-  let output = "";
+  let output: Record<string, string> = {};
 
   let operations: Record<string, { operation: OperationObject; pathItem: PathItemObject }> = {};
 
   // --raw-schema mode
-  if (rawSchema) {
-    switch (version) {
+  if (ctx.rawSchema) {
+    const required = new Set(Object.keys(schema));
+    switch (ctx.version) {
       case 2: {
-        return `export interface definitions {\n  ${transformSchemaObjMap(schema, {
-          formatter,
-          immutableTypes,
-          required: Object.keys(schema),
-          version,
-        })}\n}`;
+        output.definitions = transformSchemaObjMap(schema, { ...ctx, required });
```

Comment:
Changing `transformAll()`’s default output to be an object rather than a string is required for the upcoming #602. But I wanted to get that change in before we release v4, so that v3 patches will be easier to work on (less codebase distance between v3 and v4)